### ISSUE TITLE
Layer: expose all layer options to props

### ISSUE THREAD

**Issue by Wykks** [alex3165/react-mapbox-gl#370]
We should expose every option of layer as props instead of exposing only some of them (and then remove layerOptions prop)

**Comment by HelloQingGuo** [comment]
Does that mean the background prop of layer will be supported from V3.0.0

**Comment by alex3165** [comment]
@HelloQingGuo What do you mean by background prop? do you mean the `background-color` in the paint prop? if so it is already possible by doing this:
```
<Layer paint={{ "background-color": "your color here" }} ... />
```

**Comment by alex3165** [comment]
Released from tag `3.0.0`
